MediaTesterLib/MediaTester.cs

```diff
@@ -6,27 +6,110 @@
-	public delegate void ExceptionHandler(MediaTester mediaTester, Exception exception);
+
+
+
+
+	public class VerifiedBlockEventArgs : EventArgs
+	{
+		public VerifiedBlockEventArgs(long absoluteDataBlockIndex, long absoluteDataByteIndex, string testFilePath, long readBytesPerSecond, int bytesVerified, int bytesFailed)
+		{
+			AbsoluteDataBlockIndex = absoluteDataBlockIndex;
+			AbsoluteDataByteIndex = absoluteDataByteIndex;
+			TestFilePath = testFilePath;
+			VerifyBytesPerSecond = readBytesPerSecond;
+			BytesVerified = bytesVerified;
+			BytesFailed = bytesFailed;
+		}
+
+		public long AbsoluteDataBlockIndex;
+		public long AbsoluteDataByteIndex;
+		public string TestFilePath;
+		public long VerifyBytesPerSecond;
+		public int BytesVerified;
+		public int BytesFailed;
+	}
+
+	public class WrittenBlockEventArgs : EventArgs
+	{
+		public WrittenBlockEventArgs(long absoluteDataBlockIndex, long absoluteDataByteIndex, string testFilePath, long writeBytesPerSecond, int bytesWritten, int bytesFailedWrite)
+		{
+			AbsoluteDataBlockIndex = absoluteDataBlockIndex;
+			AbsoluteDataByteIndex = absoluteDataByteIndex;
+			TestFilePath = testFilePath;
+			WriteBytesPerSecond = writeBytesPerSecond;
+			BytesWritten = bytesWritten;
+			BytesFailedWrite = bytesFailedWrite;
+		}
+
+		public long AbsoluteDataBlockIndex;
+		public long AbsoluteDataByteIndex;
+		public string TestFilePath;
+		public long WriteBytesPerSecond;
+		public int BytesWritten;
+		public int BytesFailedWrite;
+	}
+
+	public class FileDeletedEventArgs : EventArgs
+	{
+		public FileDeletedEventArgs(int _totalFiles, int _removedFiles)
+		{
+			totalFiles = _totalFiles;
+			removedFiles = _removedFiles;
+		}
+
+		public int totalFiles;
+		public int removedFiles;
+	}
+
+	public class ExceptionEventArgs : EventArgs
+	{
+		public ExceptionEventArgs(Exception e)
+		{
+			Exception = e;
+		}
+
+		public Exception Exception;
+	}
 
 	public class MediaTester
 	{
-		private const FileOptions FileFlagNoBuffering = (FileOptions)0x20000000;
+		/// <summary>
+		/// Specify the size of each block, in bytes.
+		/// </summary>
 		public const int DATA_BLOCK_SIZE = 8 * 1024 * 1024;
-		public const int DATA_BLOCKS_PER_FILE = 1 * 1024 / 8;
-		public const int FILE_SIZE = DATA_BLOCK_SIZE * DATA_BLOCKS_PER_FILE; // 1 GiB == 1024 * 1024 * 1024 == 1,073,741,824
+
+		/// <summary>
+		/// Specify the number of blocks per files.
+		/// </summary>
+		public const int DATA_BLOCKS_PER_FILE = 128;
+
+		/// <summary>
+		/// The total size of a test file, in bytes.
+		/// </summary>
+		public const int FILE_SIZE = DATA_BLOCK_SIZE * DATA_BLOCKS_PER_FILE;
+
 		public const string TempSubDirectoryName = "MediaTester";
 
-		public WriteBlockCompleteHandler AfterWriteBlock;
-		public VerifyBlockCompleteHandler AfterQuickTest;
-		public VerifyBlockCompleteHandler AfterVerifyBlock;
-		public ExceptionHandler OnException;
```

Comment:
Why did you change all of these names? I am fine with changing from delegate to event but the names should stay the same.